components/logins/ios/Logins/LoginsStorage.swift

```diff
@@ -24,36 +24,53 @@ open class SyncUnlockInfo {
 let queue = DispatchQueue(label: "org.mozilla.sync15.logins")
 
 open class LoginsStorage {
-    var raw: OpaquePointer? = nil
+    private var raw: UInt64 = 0
```

Comment:
Hmm, so Kotlin uses an atomic, but not Swift...is that because we have dispatch queues to handle concurrent calls correctly?

Reply:
The kotlin doesn't strictly need to use the atomic either, all access is Synchronized. It's atomic mostly for convenience. After this patch we can re-evaluate if its worth synchonizing access on this side though, given that the rust side is protected by a Mutex.